File: Project/Data/Entities/Room.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

public class Room
{
    [Key]
    public int RoomId { get; set; }

    [Required]
    public int HotelId { get; set; }

    [ForeignKey("HotelId")]
    public Hotel Hotel { get; set; }

    [Required]
    public RoomType RoomType { get; set; }

    [Required]
    [StringLength(500)]
    public string Description { get; set; }

    [Required]
    [Range(0, 10000)]
    public double Price { get; set; }

    [Required]
    public bool isAvailable { get; set; }
}

File: hotel_booking_system_frontend_flutter/lib/Model/room.dart
class Room {
  final String roomType;
  final double price;
  final String description;
  final bool isAvailable;
  final bool isBooked;

  Room({
    required this.roomType,
    required this.price,
    required this.description,
    required this.isAvailable,
    required this.isBooked,
  });

  Map<String, dynamic> toJson() {
    return {
      'roomType': roomType,
      'price': price,
      'description': description,
      'isAvailable': isAvailable,
      'isBooked': isBooked,
    };
  }
}
